subprojects/android-test/ui-testing-core-app/src/androidTest/kotlin/com/avito/android/test/app/core/GodRuleChain.kt

```diff
@@ -11,28 +12,22 @@ import org.junit.runners.model.Statement
 /**
  * RuleChain with exposed methods and fields from it's rules
  */
-class GodRuleChain<out T : Activity>(private val chain: RuleChain) : TestRule {
+class GodRuleChain<T : Activity>(private val chain: RuleChain) : TestRule {
 
     @Suppress("UNCHECKED_CAST")
     private val rules = chain.javaClass.getDeclaredField("rulesStartingWithInnerMost")
         .apply { isAccessible = true }.get(chain) as List<TestRule>
 
-    private val activityTestRule = rules.filterIsInstance<androidx.test.rule.ActivityTestRule<T>>().first()
+    private val activityTestRule = rules.filterIsInstance<BaseActivityScenarioRule<T>>().first()
 
-    val activity: T
-        get() = activityTestRule.activity
+    fun onActivity(func: T.() -> Unit): ActivityScenario<T> = activityTestRule.scenario.onActivity { func(it) }
```

Comment:
Does this function have timeout or we will wait infinitely?

Reply:
Will wait for idle and perform the `func` on Activity